app/controllers/application_controller.rb

```diff
@@ -1,6 +1,26 @@
 class ApplicationController < ActionController::Base
     protect_from_forgery with: :exception
     before_action :authenticate_user!
+    include Pundit
+    rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
     # attr_reader :current_user
+    def after_sign_in_path_for(resource)
```

Comment:
`resource` variable is unused here.
try to see if you can get `resource.role` if yes the user that instead of `current_user.role`